crates/utils/src/settings/structs.rs

```diff
@@ -9,29 +9,19 @@ pub struct Settings {
   /// settings related to the postgresql database
   #[default(Default::default())]
   pub database: DatabaseConfig,
-  /// rate limits for various user actions, by user ip
-  #[default(Some(Default::default()))]
-  pub rate_limit: Option<RateLimitConfig>,
   /// Settings related to activitypub federation
-  #[default(Default::default())]
-  pub federation: FederationConfig,
   /// Pictrs image server configuration.
   #[default(Some(Default::default()))]
   pub(crate) pictrs: Option<PictrsConfig>,
-  #[default(Default::default())]
-  pub captcha: CaptchaConfig,
   /// Email sending configuration. All options except login/password are mandatory
-  #[default(None)]
-  #[doku(example = "Some(Default::default())")]
-  pub email: Option<EmailConfig>,
   /// Parameters for automatic configuration of new instance (only used at first start)
   #[default(None)]
   #[doku(example = "Some(Default::default())")]
   pub setup: Option<SetupConfig>,
   /// the domain name of your instance (mandatory)
   #[default("unset")]
   #[doku(example = "example.com")]
-  pub hostname: String,
+  pub hostname: String, // TODO this is duplicated in the instance / local_site table now tho?
```

Comment:
I suppose you can test if setup and tests still work if you remove the config field.